dcm4che-conf/dcm4che-conf-core/src/test/java/org/dcm4che3/conf/core/refindexing/RefIndexingTest.java

```diff
@@ -65,6 +64,45 @@ public void testSimpleChanges() {
 
     }
 
+    @Test
+    public void refreshNode_whenNodeIsNotAvailableInConfigurationAnymore_removesNodeFromIndex() {
+        SampleBigConf sampleBigConf = new SampleBigConf();
+
+        sampleBigConf.child1 = new SampleReferableConfClass("UUID1");
+        sampleBigConf.child1.setMyName("Romeo");
+
+        SampleReferableConfClass uuid2obj = new SampleReferableConfClass("UUID2");
+        uuid2obj.setMyName("Juliet");
+
+        sampleBigConf.childList.add(uuid2obj);
+        sampleBigConf.childList.add(new SampleReferableConfClass("UUID3"));
+
+        sampleBigConf.childMap.put("first", new SampleReferableConfClass("UUID4"));
+        sampleBigConf.childMap.put("second", new SampleReferableConfClass("UUID5"));
+
+        configuration.persistNode(new Path("confRoot"), vitalizer.createConfigNodeFromInstance(sampleBigConf), null);
+
+        // check if uuids are indexed
+        Assert.assertEquals(5, uuidToSimplePathCache.size());
```

Comment:
Any chance you can use assertThat and messages for this and asserts on lines 90+?

Reply:
Done